### ISSUE TITLE
Error Message shows more than 20 times in a row

### ISSUE THREAD

**Issue by sakura1977** [sqlitebrowser/sqlitebrowser#3161]
Details for the issue
--------------------

#### What did you do?

I have a database with a special, custom collation, which is unknown to the DB Browser.
Every time, when running a sql query or looking at data from a table, I get the following error message box showing more than 20 times.

---------------------------
Collation needed! Proceed?
---------------------------
A table in this database requires a special collation function 'SYSTEMNOCASE' that this application can't provide without further knowledge.
If you choose to proceed, be aware bad things can happen to your database.
Create a backup!
---------------------------
&Yes   &No   
---------------------------

#### What did you expect to see?

Maybe once per opened database, this message would be helpful, but neither every time, nor for every record/field encountered within the database.

#### What did you see instead?

I see that error message more than 20 times EVERY time I run a sql query or look at data.

Useful extra information
-------------------------
> DB4S v3.12.2 [built for x86_64-little_endian-llp64] on Windows 10 (10.0) (winnt/10.0.25188) [x86_64]
> using SQLite Version 3.35.5
> and Qt 5.12.8

**Comment by chrisjlocke** [comment]
This vaguely rings a bell, where a suggestion was suggested (?) that errors should be queued up and filtered rather than screaming every time they're encountered.  That approach would take some considerable development.  
I don't think there is a quick fix, apart from fork the application and knobble the message! ;)

Obvious fix #2 is to remove this collation in a temporary database, but I assume that's not possible, and certainly not a great workaround.

**Comment by MKleusberg** [comment]
Since #2356 there should be a "Yes. Don't ask again" button in this dialog and the version of DB4S you're using should include this. Don't you see this button or is it not working for you?

**Comment by sakura1977** [comment]
I checked, I'm running on version 3.12.2 - downloaded it a couple days ago.
I recall this problem starting around second half of 2019, but then changing jobs and now just returning to using DB4S. 

Looking a bit closer into it, the message box comes exactly as often, as the **first table** has the unknown collation for its fields. In my case, 21 columns use that collation, so the warning appears 21 times.

@MKleusberg  As for a **Yes. Don't ask again** button. There is none. As from the dialog content copy above, only the **Yes** and **No** button appear. 

Regards,
Daniel

**Comment by chrisjlocke** [comment]
Could you try with the nightly?
https://nightlies.sqlitebrowser.org/win32/2022-07/DB.Browser.for.SQLite-2022-07-09-win32.zip

This doesn't need installing - just extract and run the .exe.  It won't clash with the currently installed 3.12.2 version.
It may be the fix didn't get added from the development branch into the live release.

**Comment by sakura1977** [comment]
Hi,

hard to tell, whether the fix works or not. 

I do not get the warning at all anymore. So it may be that the collation type has been added to the DB Browser, or it may be, that it simply does not show the error at all anymore.

Anyways, personally, for me, it solves the problem :-) 

Regards,
Daniel

**Comment by chrisjlocke** [comment]
OK, I'll try a test later.  The only collations DB4S knows about are in this list:

![image](https://user-images.githubusercontent.com/7872218/188812601-7b20ce01-f580-45ea-99cb-fa0244bda22d.png)

So don't think anything has been 'added'.  But hey - at least you don't get a million notifications now! ;)

**Comment by sakura1977** [comment]
Yeah, there seems no new collation. So I would expect one warning message
